src/dos/drive_fat.cpp

```diff
@@ -433,7 +433,7 @@ bool fatDrive::getEntryName(char *fullname, char *entname) {
 	return true;
 }
 
-bool fatDrive::getFileDirEntry(char const * const filename, direntry * useEntry, Bit32u * dirClust, Bit32u * subEntry) {
+bool fatDrive::getFileDirEntry(char const * const filename, direntry * useEntry, Bit32u * dirClust, Bit32u * subEntry, bool dirOk) {
```

Comment:
Use `cstdint` types (and read-only, if possible), and hopefully the formatted will wrap the arguments too so they're not running off.

Reply:
I changed Bit32u to uint32_t, and made dir_ok read-only.